### PR TITLE
Add deactivate-feature feature to test validator cli

### PR DESCRIPTION
**Pull request by charlieyou**
#### Problem
Copied from #23005:

Currently the solana-test-validator indiscriminately enables all features, vis-a-vis TestValidatorGenesis genesis_config creation which results in different behavior then devnet, testnet or mainnet-beta. Transaction wide budget being a great example of confusion to program developers.

TestValidatorGenesis has been updated to allow deactivation of features: https://github.com/solana-labs/solana/pull/22860 but currently only available programmatically within user testing that use it to get the local validator running.

solana-test-validator should expose CLI options to leverage this capability to provide developer with predictable state when deploying/running programs on other clusters.

#### Summary of Changes
Adds a `-d, --deactivate-feature` CLI flag to `solana-test-validator` which passes its pubkey arguments to the `TestValidatorGenesis::deactivate_features`

Fixes (part of) #23005

#### Proof of work
`./test-install/bin/solana-test-validator -d 21AWDosvp3pBamFW91KB35pNoaoZVTM7ess8nr2nt53B`
```
solana feature status -ul
...
21AWDosvp3pBamFW91KB35pNoaoZVTM7ess8nr2nt53B | inactive                    | merge NonceError into SystemError
```

### CODE REVIEW

File: validator/src/bin/solana-test-validator.rs [hunk 1]
```diff
@@ -325,6 +325,16 @@ fn main() {
                 .long("no-accounts-db-caching")
                 .help("Disables accounts caching"),
         )
+        .arg(
+            Arg::with_name("deactivate_feature")
+                .long("deactivate-feature")
+                .short("d")
```
Comment (mvines) [inline validator/src/bin/solana-test-validator.rs:331]: This looks great, thanks. My only nit is regarding camping on the `-d` short arg. I think I'd rather leave this out and only support the long `--deactivate-feature` form for now. `-d` is quite generic and maybe there'll be a more important arg in the future we'll want to use this for (and I fully recognize I set this precedent with `-c`!)

Comment (mvines) [reply validator/src/bin/solana-test-validator.rs:331]: `-df` seems decent. I never actually tried a `.short()` arg with more than one characters in it, lol it works though?

Comment (charlieyou) [reply validator/src/bin/solana-test-validator.rs]: It does not actually work :(

**Comment by FrankC01** [comment]
Nice work @charlieyou ! Look forward to using it.

**Comment by charlieyou** [comment]
@mvines CI just passed, thanks for the heads up on that

@FrankC01 Mostly your work lol, this part ended up being teensy tiny.

**Comment by mvines** [comment]
Thanks for the PR @charlieyou!

**Comment by mvines** [comment]
@Mergifyio backport v1.9

**Pull request merged**